assignments/pull_requests.go

```diff
@@ -0,0 +1,157 @@
+//	| Test Name | Score | Weight | % of Total |
+//	| :-------- | :---- | :----- | ---------: |
+//  | Test 1	| 2/6	| 1		 |	   12.23% |
+//  | Test 2	| 1/3   | 2	     |     24.46% |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  | Total		|		|		 |	   48.23% |
+// Once a total score of 80% is reached, reviewers are automatically assigned.
+//
+func CreateFeedbackComment(results *score.Results, task *pb.Task, assignment *pb.Assignment) string {
+	body := "## Test results from latest push\n"
+	body += "| Test Name | Score | Weight | % of Total |\n"
+	body += "| :-------- | :---- | :----- | ---------: |\n"
+
+	for _, score := range results.Scores {
+		if score.TaskName != task.LocalName() {
+			continue
+		}
+		percentageScore := (float64(score.Score) / float64(score.MaxScore)) * (float64(score.Weight) / results.TotalTaskWeight(task.LocalName()))
+		body += fmt.Sprintf("| %s | %d/%d | %d | %.2f%% |\n", score.TestName, score.Score, score.MaxScore, score.Weight, percentageScore*100)
+	}
+	// TODO(espeland): TaskSum retruns an int, while a float is used for individual tests
+	body += fmt.Sprintf("| **Total** | | | **%d%%** |\n\n", results.TaskSum(task.LocalName()))
+	body += fmt.Sprintf("Once a total score of %d%% is reached, reviewers are automatically assigned.\n", assignment.GetScoreLimit())
+	return body
+}
+
+// AssignReviewers assigns reviewers to a group repository pull request.
+// It assigns one other group member and one course teacher as reviewers.
+func AssignReviewers(ctx context.Context, sc scm.SCM, db database.Database, course *pb.Course, repo *pb.Repository, pullRequest *pb.PullRequest) error {
+	teacherReviewer, err := getNextTeacherReviewer(db, course)
+	if err != nil {
+		return err
+	}
+	// TODO(espeland): Remember to uncomment when finished testing
+	// studentReviewer, err := getNextStudentReviewer(db, repo.GetGroupID(), pullRequest.GetUserID())
+	// if err != nil {
+	// 	return err
+	// }
+
+	reviewers := []string{
+		teacherReviewer.GetLogin(),
+		// studentReviewer.GetLogin(),
+	}
+	opt := &scm.RequestReviewersOptions{
+		Organization: course.GetOrganizationPath(),
+		Repository:   repo.Name(),
+		Number:       int(pullRequest.GetNumber()),
+		Reviewers:    reviewers,
+	}
+
+	if err := sc.RequestReviewers(ctx, opt); err != nil {
+		return err
+	}
+	// Change pull request stage to review
+	pullRequest.SetReview()
+	return db.UpdatePullRequest(pullRequest)
+}
+
+// getNextReviewer gets the next reviewer from either teacherReviewCounter or studentReviewCounter,
+// based on whoever in total has been assigned to the least amount of pull requests.
+// It is simple, and does not account for how many current review requests any user has.
+//
+// Returns an error if the list of users is empty.
+func getNextReviewer(ID uint64, users []*pb.User, reviewCounter map[uint64]map[uint64]int) (*pb.User, error) {
```

Comment:
I think this would be much cleaner if you instead pass in a single `map[uint64]int` and do the `id` lookup and map initialization from the call site, e.g., maybe using something like:

```go
type countMap map[uint64]map[uint64]int

func (m countMap) newMap(id uint64) {
	m[id] = make(map[uint64]int)
}
```

Then you don't need the `id` variable (remember to use lower case for unexported variables).